### ISSUE TITLE
Oil generating incorrectly in oceans

### ISSUE THREAD

**Issue by LuigiHutch** [BuildCraft/BuildCraft#3229]
BC 7.2.0
New world generated.

When you find a oil spout in a ocean, there is oil above the water and there is water below the ocean in the pocket but the centre column is filled with water and not oil. This means that the pump can only reach the oil above the ocean but not the large pocket below it

**Comment by AlexIIL** [comment]
Technically because the water can flow into oil, not because we have a generation bug :)

**Comment by asiekierka** [comment]
Yes, but this severely breaks pump expectations. This is bad.

**Comment by AlexIIL** [comment]
I've already fixed it in master, I was just making sure we knew that the issue wasn't with the generation code.

**Comment by MKHenson** [comment]
Hey guys, has this fix rolled yet? I can confirm the issue still exists in 7.2.3 (not tried 7.2.4 yet)

**Comment by AlexIIL** [comment]
Oh its still in 7.2.3? Was the world created in an earlier version than that?

**Comment by AlexIIL** [comment]
(In my dev environment oil generates properly)

**Comment by MKHenson** [comment]
Yeah I created a new world, and the ocean sprouts had the same issue you described. I then updated the world to 1.7.4 and the oil wells were replaced by machine blocks (Cant remember which ones they were specifically). I then deleted the world and re-created a new one in 1.7.4, and can confirm that the wells seemed to be fixed. 

Just for your reference, there are no other mods installed and my OS is Windows 10. 

But it does seem to be fixed in 1.7.4 :)
